### PR TITLE
WebGLRenderer: Add support for morphing colors.

### PR DESCRIPTION
**Pull request by Mugen87**
Related issue: #23467

**Description**

This PR adds support for morphing vertex colors. Meaning having `geometry.morphAttributes.color` is now valid.

The PR also ensures that `GLTFLoader` can load assets which define morph targets for `COLOR_0`.  

The new feature is only supported with WebGL 2 since the implementation requires the morph texture for managing the data.

### CODE REVIEW

**Comment by Mugen87** [comment]
@AleksandarCebov We can only add support for `COLOR_0` since `three.js` only support a single set of vertex colors.

**Comment by AleksandarCebov** [comment]
That is also fine. As much data that can consume as possible.

**Comment by donmccurdy** [comment]
Great work, and thanks @Mugen87 and @AleksandarCebov!

Thoughts, in no particular order ...

- Morphing vertex colors and UVs seems like a very useful technique, and _with a good authoring workflow_ I expect artists and devs will do interesting things with it.
- Popular workflows do not support the technique today. I checked Blender, Maya, and Unreal Engine, and all appear to support only positions and normals in morph targets.
- If we were to pick only one of the two (morph colors vs morph UVs) I might choose UVs, since that provides the option of morphing colors on a simple gradient, or more complex animations like flipboards or scrolling textures. Of course we could also do both.
- Node-based shaders can give similar results without the need of a dedicated morph target feature within the renderer. Usability might be a bit lower, e.g. users must identify the right input node to morph the colors, rather than accessing `mesh.morphTargetInfluences`. And I'm not sure of the right timeline for providing node-based materials from GLTFLoader.


tl;dr — We'd be stepping out a bit "in front" of the ecosystem by adding this feature before real workflows exist. There are times when that's a good thing to do, so I'm not opposed to adding this, but just want to be sure we're aware of both the implications and alternatives.

**Comment by mrdoob** [comment]
> Didn't know ro.me used this method! Do you recall what the Blender workflow was? I guess it was using three.js' custom Blender exporter at that point...

Yes, we had a bunch of python scripts I think.

**Comment by donmccurdy** [comment]
Just tested this out, working other than a minor issue fixed in #23654!

1. Export from Blender with N morph targets and N+1 vertex color sets (base color + 1 for each target, in that order)
2. Open model in https://gltf.report/
3. Paste [script](https://gist.github.com/donmccurdy/d946c4a1fab1132be19c6495835cbad1) into script tab, and execute
4. Export

https://user-images.githubusercontent.com/1848368/156679038-bc4929ed-caae-4162-bf88-ade3a6ac89f9.mov
